witness/src/Dscp/Witness/Web/Types.hs

```diff
@@ -107,6 +113,9 @@ data HashIs
 deriving instance (Eq (Id a), Eq a) => Eq (PaginatedList d a)
 deriving instance (Show (Id a), Show a) => Show (PaginatedList d a)
 
+instance Buildable ByteString where
+    build _ = "<binary data, omitted>"
+
```

Comment:
I mean, I still think it's important to create a dedicated newtype and use it `checkFairCVPDF` (maybe even `Pdf.PDFBody` itself).